pkg/forwarder/forwarder.go

```diff
@@ -152,6 +152,24 @@ type Forwarder interface {
 // Compile-time check to ensure that DefaultForwarder implements the Forwarder interface
 var _ Forwarder = &DefaultForwarder{}
 
+// Options contain the configuration options for the DefaultForwarder
+type Options struct {
+	NumberOfWorkers      int
+	RetryQueueSize       int
+	EnableHealthChecking bool
+	KeysPerDomain        map[string][]string
+}
+
+// NewOptions creates new Options with default values
+func NewOptions(keysPerDomain map[string][]string) *Options {
+	return &Options{
+		NumberOfWorkers:      config.Datadog.GetInt("forwarder_num_workers"),
+		RetryQueueSize:       config.Datadog.GetInt("forwarder_retry_queue_max_size"),
+		EnableHealthChecking: true,
```

Comment:
I know that currently the Forwarder only check the API key validity. But I think that having only a On/Off for the HealthChecking is not futur proof.
what do you think to rename the option: `EnableHealthChecking` to `DisableAPIKeyChecking` which mean that by default the API Key check is enabled (backwardcompatibility)
```golang
type Options struct {
	NumberOfWorkers       int
	RetryQueueSize        int
	DisableAPIKeyChecking bool
	KeysPerDomain         map[string][]string
}
```
and forward the option to the `forwarderHealth{}` instantiation or a new method `NewForwarderHealth()`

```

func NewDefaultForwarder(options *Options) *DefaultForwarder {
	f := &DefaultForwarder{
		NumberOfWorkers:  config.Datadog.GetInt("forwarder_num_workers"),
		NumberOfWorkers:  options.NumberOfWorkers,
		domainForwarders: map[string]*domainForwarder{},
		keysPerDomains:   map[string][]string{},
		internalState:    Stopped,
		healthChecker:   NewForwarderHealth(options),
	}
```

Reply:
Does it need to be future proof?  I don't believe this is intended to be used external to the agent.

Reply:
when I say future proof. it is for example adding other check that only the API key validity, for example the size of a queue.
This PR deactivate completely the heath check, and introduce some logic to verify that the pointer is not nil.